BedrockServer.cpp

```diff
@@ -897,10 +906,27 @@ void BedrockServer::worker(SData& args,
                             // to the minimum time required.
                             bool commitSuccess = false;
                             {
+                                // Don't increment if we're blocking, we've already done it when we set the flag.
+                                if (!blocking) {
+                                    // Increment this *before* we lock, we want to know how many are waiting, only one will
+                                    // ever actually have the lock.
+                                    int64_t newPendingCount =  ++server._pendingCommitCount;
+                                    if (newPendingCount > server._maxPendingCommits.load()) {
```

Comment:
does this mean that we will only ever be able to have `maxPendingCommits` worth of commands being written at the same time? which by default would mean we've gone from 192 to 5?

Reply:
Looks that way to me too, @coleaeason, though the comments here https://github.com/#issuecomment-440253081 in the description and the pseudo code suggest otherwise. @tylerkarasweski can you clarify what you intend this to be doing, I.e. are we expecting to have unlimited concurrent writes as long as we’re not conflicting? Or are we limiting concurrent commit-ready threads to 5 at any one time?

Reply:
This stops new writes from starting when there's already a line of commands waiting for the commit lock. It does not limit the number of simultaneous writes, so long as this line remains short. The max number of simultaneous commits remains 1, as it always has been.

As a real-world analogy, this is like walking into Costco, and looking at the checkout line. If there are more than five people waiting, you walk out and decide to come back and shop later. It doesn't stop 192 people from shopping at once, so long as the checkout line stays shorter than 5.

(In this analogy "shopping" is writing, and "checking out" is committing. Not sure if that analogy clarifies at all)

The analogy breaks down with conflicts, but the idea is that the number of conflicts are reduced if we don't start writing commands when we expect we'll just end up waiting in a line to commit them anyway. Since we didn't do the write during the congested time, hopefully the risk of conflict drops.